app/controllers/patreon_users_controller.rb

```diff
@@ -1,65 +1,87 @@
+    current_user.patreon.try(:destroy)
+    redirect_to settings_path, notice: 'Patreon account unlinked :)'
+  end
 
-    api_client = Patreon::API.new(access_token)
-    user_response = api_client.fetch_user
+  private
 
-    user = user_response['data']
-    if user.nil?
-      redirect_to settings_path, alert: 'There was an issue connecting your Patreon account. Please try again.'
-      return
-    end
+  def patreon_oauth_client
+    Patreon::OAuth.new(ENV['PATREON_CLIENT_ID'], ENV['PATREON_CLIENT_SECRET'])
+  end
 
-    included = user_response['included']
-    pledge_cents = 0
-    pledge_created_at = nil
+  def patreon_api_client(access_token)
+    Patreon::API.new(access_token)
+  end
 
-    if included
-      pledge = included.find do |obj|
-        obj['type'] == 'pledge' && obj['relationships']['creator']['data']['id'] == creator_id
-      end
+  def redirect_uri
+    "#{request.protocol}#{request.host_with_port}/auth/patreon/callback"
+  end
 
-      pledge_cents = pledge['attributes']['amount_cents']
-      pledge_created_at = pledge['attributes']['created_at']
-    end
+  def update_patreon_user(user:, access_token:, refresh_token:)
+    user_response = patreon_api_client(access_token).fetch_user
+    return nil if user_response['data'].nil?
 
-    PatreonUser.find_or_initialize_by(user: current_user).update(
+    PatreonUser.find_or_initialize_by(user: user).update(
       access_token:  access_token,
       refresh_token: refresh_token,
 
-      patreon_id: user['id'],
-      full_name:  user['attributes']['full_name'],
+      patreon_id: user_response['data']['id'],
+      full_name:  user_response['data']['attributes']['full_name'],
 
-      pledge_cents:      pledge_cents,
-      pledge_created_at: pledge_created_at
+      pledge_cents:      pledge_cents(user_response),
+      pledge_created_at: pledge_created_at(user_response)
     )
+  end
 
-    redirect_to(
-      settings_path,
-      notice: "You're now linked with Patreon! If you're a patron, your benefits have been activated :)"
-    )
+  def creator_id
+    '2972755'
   end
 
-  def unlink
-    current_user.patreon.try(:destroy)
+  def pledge_cents(user_response)
+    return 0 if user_response['included'].blank?
 
-    redirect_to settings_path, notice: 'Patreon account unlinked :)'
+    user_response['included'].find do |obj|
+      obj['type'] == 'pledge' && obj['relationships']['creator']['data']['id'] == creator_id
+    end['attributes']['amount_cents']
   end
 
-  private
+  def pledge_created_at(user_response)
+    return nil if user_response['included'].blank?
 
-  def redirect_uri
-    "#{request.protocol}#{request.host_with_port}/auth/patreon/callback"
+    user_response['included'].find do |obj|
+      obj['type'] == 'pledge' && obj['relationships']['creator']['data']['id'] == creator_id
+    end['attributes']['created_at']
+  end
+
+  def link_success_msg
+    'You\'re now linked with Patreon! If you\'re a patron, your benefits have been activated :)'
```

Comment:
Instead of using `\'` for the quotes, it should probably just be wrapped with regular quotes.  Rubocop won't throw a cop warning if there is a apostrophe in the string in the quotes.

Reply:
Ohh cool I didn't know that about Rubocop!